### PR TITLE
Add Interface for content extractors

### PR DESCRIPTION
**Pull request by KomalBhadani27**
Added interface for content extractor classes

### CODE REVIEW

File: extraction/extractor.py [hunk 1]
```diff
@@ -72,13 +72,13 @@ def __extract_data_from_html_head(self, soup_head):
     def __extract_data_from_html_body(self, node):
         """This function iterates over dom using dfs"""
 
-        is_content, _content = self.__is_content(node)
+        _content = self.__is_content(node)
```
Comment (detel) [inline extraction/extractor.py:75]: `__is_content(node)` to `__validate_and_extract_content(node)`

Comment (KomalBhadani27) [reply extraction/extractor.py]: Updated!

File: extraction/extractor.py [hunk 2]
```diff
@@ -72,21 +72,21 @@ def __extract_data_from_html_head(self, soup_head):
     def __extract_data_from_html_body(self, node):
         """This function iterates over dom using dfs"""
 
-        is_content, _content = self.__is_content(node)
+        _content = self.__validate_and_extract_content(node)
 
         # Add to the final list if a valid content is extracted
```
Comment (detel) [inline extraction/extractor.py:77]: early return
```
if not _content:
     return

self.contents_list.add_content(_content)
if node and node.name:
            for child in node.contents:
                self.__extract_data_from_html_body(child)
```

Comment (KomalBhadani27) [reply extraction/extractor.py:77]: Updated in next iterations!

**Pull request merged**